src/zenml/artifacts/artifact_config.py

```diff
@@ -83,3 +91,35 @@ def _remove_old_attributes(cls, data: Dict[str, Any]) -> Dict[str, Any]:
             )
 
         return data
+
+    @model_validator(mode="after")
+    def artifact_config_after_validator(self) -> "ArtifactConfig":
+        """Artifact config after validator.
+
+        Returns:
+            The artifact config.
+        """
+        if isinstance(self.name, str):
+            _name = format_name_template(self.name)
+            self._is_dynamic = _name != self.name
+            self.name = _name
+        elif callable(self.name):
+            self.name = self.name()
+            self._is_dynamic = True
+        return self
+
+    @property
+    def _evaluated_name(self) -> Optional[str]:
+        """Evaluated name of the artifact.
+
+        Returns:
+            The evaluated name of the artifact.
+
+        Raises:
+            RuntimeError: If the name is still a callable.
+        """
+        if callable(self.name):
```

Comment:
Do we want to reuse the date/time from the pipeline run here, same as we do with model version names? If it's even possible to pass this here, I'm not sure

Reply:
Might be interesting. Maybe this can be set on a context level instead of pulling fresh each time.

Reply:
I didn't implement this, but I can give it a shot if the PR is conceptually fine with you.